clients/privacy-center/cypress/e2e/consent-events.cy.ts

```diff
@@ -0,0 +1,640 @@
+      const events = stub.getCalls().map((call) => ({
+        type: call.args[0].type,
+        detail: call.args[0].detail as FidesEventDetail,
+      }));
+
+      // For each event, verify only the properties we care about
+      events.forEach((actual, index) => {
+        const { type, detail: expectedDetail } = expectedEvents[index];
+        expect(actual.type, `Event ${index} type`).to.equal(type);
+        expect(actual.detail, `Event ${index} detail`).to.be.an("object");
+
+        // Verify timestamp exists and is a valid performance.now() value
+        expect(actual.detail.timestamp).to.be.a("number");
+
+        // Only verify the properties we specified in expectedEvents
+        Object.entries(expectedDetail).forEach(([key, value]) => {
+          // First verify the key exists if we expect a value
+          if (Object.keys(value).length > 0) {
+            expect(actual.detail).to.have.property(key);
+          }
+          expect(
+            actual.detail[key],
+            `Event ${index} detail.${key}`,
+          ).to.deep.include(value);
+        });
+      });
+
+      // Verify the sequence of events matches exactly
+      expect(
+        events.map((e) => e.type),
+        "Expected event types to match exactly in order",
+      ).to.deep.equal(expectedEvents.map((e) => e.type));
+    });
+  }
+
+  describe("when banner_and_modal experience", () => {
+    beforeEach(() => {
+      // Load the banner_and_modal experience
+      const fixture = "experience_banner_modal.json";
+      cy.fixture(`consent/${fixture}`).then((data) => {
+        let experience = data.items[0];
+        stubConfig({ experience });
+      });
+    });
+
+    it("should fire FidesEvents for all key interactions", () => {
+      const expectedEvents: FidesEventExpectation[] = [];
+
+      // Initialize and show banner on page load
+      expectedEvents.push(
+        { type: "FidesInitializing", detail: {} },
+        { type: "FidesInitialized", detail: {} },
+        {
+          type: "FidesUIShown",
+          detail: { extraDetails: { servingComponent: "banner" } },
+        },
+      );
+
+      // Open modal from banner button
+      cy.get("#fides-banner .fides-manage-preferences-button").click();
+      expectedEvents.push({
+        type: "FidesUIShown",
+        detail: { extraDetails: { servingComponent: "modal" } },
+      });
+
+      // Toggle first notice on then off
+      cy.get("#fides-modal .fides-toggle-input").first().click().click();
+      expectedEvents.push(
+        {
+          type: "FidesUIChanged",
+          detail: {
+            extraDetails: {
+              servingComponent: "modal",
+              trigger: {
+                type: "toggle",
+                label: "Advertising",
+                checked: true,
+              },
+              preference: {
+                key: "advertising",
+                type: "notice",
+              },
+            },
+          },
```

Comment:
See? This event is rich and useful, it has both the UI context from the `trigger` and the business-logic context from the `preference`.